app/models/organizer_position.rb

```diff
@@ -55,11 +55,20 @@ def tourable_options
   end
 
   def self.role_at_least?(user, event, role)
-    return false unless event.present? && role.present?
-    return true if user&.admin?
+    if role.to_s == "reader"
+      return event.ancestor_organizer_positions.reader_access.where(user:).exists?
+    end
 
-    current = find_by(user:, event:)&.role
-    current && roles[current] >= roles[role]
+    if role.to_s == "member"
+      # Only check direct organizer positions, unless the user is a manager of an ancestor
+      return event.organizer_positions.member_access.where(user:).exists? || event.ancestor_organizer_positions.member_access.where(user:).exists?
```

Comment:
please double check if it's a typo

```suggestion
      return event.organizer_positions.member_access.where(user:).exists? || event.ancestor_organizer_positions.reader_access.where(user:).exists?
```